packages/vscode-extension/src/devices/DeviceNameCache.ts

```diff
@@ -0,0 +1,141 @@
+import path from "path";
+import { workspace, Uri } from "vscode";
+import { extensionContext } from "../utilities/extensionContext";
+import { Logger } from "../Logger";
+
+type DeviceModel = {
+  brand: string;
+  name: string;
+};
+type DeviceModels = Record<string, DeviceModel>;
+
+type CacheFile = {
+  savedAt: number;
+  data: DeviceModels;
+};
+
+const DEVICE_MODELS_URL = "https://cdn.jsdelivr.net/gh/bsthen/device-models/devices.json";
+const DEVICE_MODELS_FILENAME = "RNIDE_device_models.json";
+const CACHE_DURATION_MS = 5 * 24 * 60 * 60 * 1000;
+
+async function readCacheFile(): Promise<CacheFile | null> {
+  try {
+    const filePath = path.join(extensionContext.globalStorageUri.fsPath, DEVICE_MODELS_FILENAME);
+    const fileUri = Uri.file(filePath);
+    const file = await workspace.fs.readFile(fileUri);
+    const json = JSON.parse(file.toString()) as CacheFile;
+    return json;
+  } catch (error) {
+    if ((error as NodeJS.ErrnoException)?.code === "ENOENT") {
+      Logger.info("Device models cache file does not exist, a new one will be created");
+      return null;
+    }
+    Logger.error(`Error reading device models cache file: ${(error as Error).message}`);
+    return null;
+  }
+}
+
+async function writeCacheFile(data: DeviceModels): Promise<void> {
+  try {
+    const filePath = path.join(extensionContext.globalStorageUri.fsPath, DEVICE_MODELS_FILENAME);
+    const contents = {
+      savedAt: Date.now(),
+      data: data,
+    };
+    await workspace.fs.createDirectory(Uri.file(path.dirname(filePath)));
+    await workspace.fs.writeFile(
+      Uri.file(filePath),
+      Buffer.from(JSON.stringify(contents), "utf-8")
+    );
+    return;
+  } catch (error) {
+    Logger.error(`Error writing device models cache file: ${(error as Error).message}`);
+  }
+}
+
+async function fetchDeviceModels(): Promise<DeviceModels> {
+  try {
+    const response = await fetch(DEVICE_MODELS_URL);
+    const json = await response.json();
+    return json;
+  } catch (error) {
+    Logger.error(`Error fetching device models: ${(error as Error).message}`);
+    return {};
+  }
```

Comment:
I'd either `throw` or return `undefined` here, so that we don't write the empty object to cache (and try to refetch instead)